presto-main/src/main/java/com/facebook/presto/operator/aggregation/multimapagg/SingleMultimapAggregationState.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.multimapagg;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.Type;
+import org.openjdk.jol.info.ClassLayout;
+
+import static com.facebook.presto.type.TypeUtils.expectedValueSize;
+
+public class SingleMultimapAggregationState
+        implements MultimapAggregationState
+{
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(SingleMultimapAggregationState.class).instanceSize();
+    private static final int EXPECTED_ENTRIES = 10;
+    private static final int EXPECTED_ENTRY_SIZE = 16;
+    private final Type keyType;
+    private final Type valueType;
+    private BlockBuilder keyBlockBuilder;
+    private BlockBuilder valueBlockBuilder;
+
+    public SingleMultimapAggregationState(Type keyType, Type valueType)
+    {
+        this.keyType = keyType;
+        this.valueType = valueType;
+        keyBlockBuilder = this.keyType.createBlockBuilder(null, EXPECTED_ENTRIES, expectedValueSize(keyType, EXPECTED_ENTRY_SIZE));
+        valueBlockBuilder = this.valueType.createBlockBuilder(null, EXPECTED_ENTRIES, expectedValueSize(valueType, EXPECTED_ENTRY_SIZE));
+    }
+
+    @Override
+    public void add(Block key, Block value, int position)
+    {
+        keyType.appendTo(key, position, keyBlockBuilder);
+        valueType.appendTo(value, position, valueBlockBuilder);
+    }
+
+    @Override
+    public void forEach(MultimapAggregationStateConsumer consumer)
+    {
+        for (int i = 0; i < keyBlockBuilder.getPositionCount(); i++) {
+            consumer.accept(keyBlockBuilder, valueBlockBuilder, i);
+        }
+    }
+
+    @Override
+    public boolean isEmpty()
+    {
+        return keyBlockBuilder.getPositionCount() == 0;
+    }
+
+    @Override
+    public int getEntryCount()
+    {
+        return valueBlockBuilder.getPositionCount();
```

Comment:
nit: why use `valueBlockBuilder` instead of `keyBlockBuilder` here? 
Being consistent across the code makes easier to understand and maintain.